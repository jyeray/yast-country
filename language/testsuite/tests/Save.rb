# encoding: utf-8

# tests for Language::Save
# Author:	jsuchome@suse.cz
# $Id$
module Yast
  class SaveClient < Client
    def main
      # testedfiles: Language.ycp

      Yast.include self, "testsuite.rb"

      @READ = {
        "sysconfig" => {
          "language" => { "RC_LANG" => "en_US.UTF-8", "RC_LC_MESSAGES" => "" }
        },
        "target"    => {
          "bash_output" => {"exit" => 0},
          "size"        => 1,
          "yast2"       => {},
          "dir"         => []
        }
      }

      TESTSUITE_INIT([@READ, {}, @READ], nil)

      Yast.import "Language"

      Language.languages = "en_US"

      TEST(lambda { Language.Save }, [@READ, {}, @READ], nil)

      nil
    end
  end
end

Yast::SaveClient.new.main
