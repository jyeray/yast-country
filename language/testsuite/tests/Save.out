Read	.target.size "/etc/sysconfig/language" 1
Read	.target.dir "/usr/share/YaST2/data/languages" []
Write	.sysconfig.language.INSTALLED_LANGUAGES "en_US" true
Write	.sysconfig.language nil true
Execute	.target.bash_output "/usr/bin/localectl set-locale LANG=en_US.UTF-8" $["exit":0]
Return	nil
